authors/apps/articles/views.py

```diff
@@ -63,6 +64,10 @@ def retrieve(self, request, slug):
             article = Article.objects.get(slug=slug)
             serializer = self.serializer_class(
                 article, context={'request': request})
+            if article.author.pk != request.user.pk:
+                username = User.objects.get(username=request.user.username)
+                ReadingStat.objects.get_or_create(
+                    user=username, articles=article)
```

Comment:
ReadingStat.objects.get_or_create(user=request.user, articles=article) produces the same result but with less code